### PR TITLE
262 add a version option at launch

### PR DESCRIPTION
**Pull request by sydhds**
Edit @thomas-senechal : 
I remade the implementation following [@gregLibert comment](https://github.com/massalabs/thyra/pull/273#pullrequestreview-1189535341). 
I also remade the flags handling following a meeting with @gregLibert .

### CODE REVIEW

**Review by thomas-senechal — APPROVED** [review]
Very good work !

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -91,6 +91,9 @@ jobs:
         with:
           go-version: ${{ env.GO_VERSION }}
       - uses: actions/checkout@v3
+      - name: Update version
+        run: |
+          echo -e "package api\n\nconst version_str=\"${GITHUB_REF_NAME}\"" > int/api/server_version.go
```
Comment (Thykof) [inline .github/workflows/CI.yml:96]: should we do this also in CD workflow?

**Review by Thykof — APPROVED** [review]

File: .github/workflows/CI.yml (hunk 1 above)
Comment (thomas-senechal) [reply .github/workflows/CI.yml:96]: You are right. I missed that. I thought we were using the CI workflow in the CD one to build the binaries. But we don't.

**Review by gregLibert — CHANGES_REQUESTED** [review]
:no_good: 
I'm not sure we are using the right method to do this:
- we are writing go code from the shell when we could have used:
  * an ldflags as explained [here](https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications), or
  * an embedded file as explained [here](https://levelup.gitconnected.com/a-better-way-than-ldflags-to-add-a-build-version-to-your-go-binaries-2258ce419d2d)
- we lose readability by moving all other server commands to an else branch
- We duplicate code such as `defer (*app).Quit()`.

My feeling is that we are adding legacy code here when we should start cleaning things up by following a boyscout approach...

As we are not in the hackathon rush anymore, can someone from the innoteam can take it over?

File: int/api/server.go [hunk 2]
```diff
@@ -18,31 +17,28 @@ import (
...
+	Port              int
+	TLSPort           int
+	TLSCertificate    string
+	TLSCertificateKey string
+	MassaNodeServer   string
+	Version           bool
+}
 
-	flag.Parse()
+func parseFlags(server *restapi.Server, startFlags StartServerFlags) {
```
Comment (Thykof) [inline int/api/server.go:29]: I think it's confusing to have 2 parseFlags function, we should name them differently

**Review by Thykof — CHANGES_REQUESTED** [review]

File: int/api/server.go (hunk 2 above)
Comment (peterjah) [reply int/api/server.go:29]: i guess it is more something about configuring the server rather than parsing flag

File: int/api/server.go [hunk 3]
```diff
@@ -60,10 +56,23 @@ func parseNetworkFlag(massaNodeServerPtr *string) {
 	os.Setenv("MASSA_NODE_URL", *massaNodeServerPtr)
 }
 
+func stopServer(app *fyne.App, server *restapi.Server, manager *pluginmanager.PluginManager) {
+	defer func() {
```
Comment (peterjah) [inline int/api/server.go:60]: I think the "defer" is useless here.. just stops the server directly

Comment (peterjah) [reply int/api/server.go:60]: defer is used to set a callback when the function returns. here your fuction does not do anything else

File: int/api/server.go [hunk 4]
```diff
@@ -103,29 +110,20 @@ func StartServer(app *fyne.App) {
...
-	parseFlags(server)
+	setAPIFlags(server, startFlags)
 
 	// Run plugins
 	manager, err := pluginmanager.New(server.Port, server.TLSPort)
 	if err != nil {
 		log.Fatalln(err)
 	}
 
-	defer func() {
```
Comment (Thykof) [inline int/api/server.go:114]: I think here we should do 

```
defer stopServer(app, server, manager)
```

File: int/api/server.go (hunk 3 above)
Comment (Thykof) [reply int/api/server.go]: I disagree, see my comment, defer is useful to stop the server if the code panic while handling an error

Comment (thomas-senechal) [reply int/api/server.go]: Oh okay. Then I will do as you said in your comment. Thanks a lot !

**Review by Thykof — APPROVED** [review]

**Review by Thykof — APPROVED** [review]
🤷 LGTM

**Review by peterjah — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]
:see_no_evil:

**Pull request merged**